cypress/page/filling-form.page.ts

```diff
@@ -0,0 +1,133 @@
+            cy.get(this.dateOfBirthSelector).type(`{selectall} ${personalInformation['dateOfBirth']}`)
+            cy.get(this.mobileNumberSelector).type(personalInformation['mobileNumber'])
+            hobbiesValues.forEach((item) => {
+                cy.get(this.hobbiesSelector).find(`[value='${item}']`).click({force: true})
+            })
+            cy.get(this.currentAddressSelector).type(personalInformation['currentAddress'])
+            this.checkStateCity(personalInformation['state'],personalInformation['city'])
+
+
+    }
+
+    private checkStateCity(state: string,city: string): void{
+        let arrowPressing = "{downArrow}" 
+        let arrowPressingCity = "{downArrow}"
+            switch(state){
+                case "Uttar Pradesh":
+                    arrowPressing = arrowPressing.repeat(2);
+                    arrowPressingCity = this.checkCityUP(city,arrowPressingCity);
+                    break;
+                case "Haryana":
+                    arrowPressing = arrowPressing.repeat(3);
+                    arrowPressingCity = this.checkCityH(city,arrowPressingCity);
+                    break;
+                case "Rajasthan":
+                    arrowPressing = arrowPressing.repeat(4);
+                    arrowPressingCity = this.checkCityR(city,arrowPressingCity);
+                    break;
+                default:
+                    arrowPressingCity = this.checkCityNCR(city,arrowPressingCity);
+
+            }
+            cy.get(this.stateSelector).click({force: true}).type(`${arrowPressing}{enter}`)
+            cy.get(this.citySelector).click({force: true}).type(`${arrowPressingCity}{enter}`)
+    }
+
+    private checkCityNCR(city: string, arrowPressing: string): string{
+        switch(city){
+            case "Gurgaon":
+                arrowPressing = arrowPressing.repeat(2);
+                break;
+            case "Noida":
+                arrowPressing = arrowPressing.repeat(3);
+                break;
+        }
+        return arrowPressing;
+    }
+
+    private checkCityUP(city: string, arrowPressing: string): string{
+            switch(city){
+                case "Lucknow":
+                    arrowPressing = arrowPressing.repeat(2);
+                    break;
+                case "Merrut":
+                    arrowPressing = arrowPressing.repeat(3);
+                    break;
+            }
+            return arrowPressing;
+    }
+
+    private checkCityH(city: string,arrowPressing: string): string{
+        switch(city){
+            case "Karnal":
+                arrowPressing = arrowPressing.repeat(2);
+                break;
+        }
+        return arrowPressing;
+    }
+                
+    private checkCityR(city: string,arrowPressing: string): string{
+        switch(city){
+            case "Jaiselmer":
+                arrowPressing = arrowPressing.repeat(2);
+                break;
+        }
+        return arrowPressing;
+    }
```

Comment:
changing the logic in `checkStateCity` these four methods (checkCityNCR, checkCityUP, checkCityH, checkCityR) are not needed.